src/buffer/out/AttrRow.cpp

```diff
@@ -429,12 +429,8 @@ void ATTR_ROW::ReplaceAttrs(const TextAttribute& toBeReplacedAttr, const TextAtt
         //      We need to fix this up below so it says G2 instead to leave room for the Y3 to fit in
         //      the new/final run.
 
-        // Copying above advanced the pointer to an empty cell beyond what we copied.
-        // Back up one cell so we can manipulate the final item we copied from the existing run to the new run.
-        pNewRunPos--;
-
         // Fetch out the length so we can fix it up based on the below conditions.
-        size_t length = pNewRunPos->GetLength();
+        size_t length = newRun.back().GetLength();

```

Comment:
since we use `newRun.back()` 3 times, should we just cache a ref to it now that we're done inserting?

```suggestion
		auto& lastRun{ newRun.back() };
        size_t length = lastRun.GetLength();
```

I know a good optimizing compiler will see right through it, but.. once I saw the third one the warning bells started going off :)

Reply:
Meh, using `back()` does make the code clearer because you also do the same thing below but with a different _back_. Hmm.

Reply:
Careful, don't commit that suggestion from the web. There's tabs in it.

Reply:
Also, I think it's fine to use `.back()` several times. This algorithm already has a lot of locals to mentally juggle. If the optimizer will see through this anyway... I'd prefer it stay contextual like this.